src/commands/gate/evaluate.ts

```diff
@@ -143,7 +149,42 @@ export class GateEvaluateCommand extends Command {
       })
   }
 
-  private handleEvaluationSuccess(evaluationResponse: EvaluationResponse): number {
+  /**
+   * Evaluate rules with retry
+   * then if service returns status:wait, retry for wait_time_ms
+   * if service returns 5xx, exponential backoff retry
+   */
+  private async evaluateRulesWithWait(
+    api: APIHelper,
+    evaluateRequest: Payload,
+    attempt?: number
+  ): Promise<AxiosResponse<EvaluationResponsePayload>> {
+    return new Promise((resolve, reject) => {
+      api
+        .evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
+        .then((response) => {
+          if (response.data.data.attributes.status === 'wait') {
+            this.context.stdout.write(renderWaiting())
+            setTimeout(() => {
+              reject(new Error('wait'))
+            }, response.data.data.attributes.metadata.wait_time_ms)
+          } else {
+            resolve(response)
+          }
+        })
+        .catch((err) => {
+          setTimeout(() => {
+            reject(err)
+          }, this.getDelay(attempt ?? 1))
+        })
+    })
+  }
+
+  private getDelay(attempt: number): number {
+    return attempt * this.initialRetryMs
```

Comment:
this is not exponential but linear with the attempt. I'd say it's ok, but let's not say exponential in the comment. If you actually want exponential you may do `2 ** attempt * this.initialRetryMs`

Reply:
I'll make it exponential, but change `initialRetryMs` to 1s, because it's gonna be waiting a lot as is.